src/shared_modules/indexer_connector/src/indexerConnector.cpp

```diff
@@ -292,36 +291,36 @@ void IndexerConnector::initialize(const nlohmann::json& templateData,
     };
 
     // Define the success callback
-    auto onSuccess = [](const std::string&)
+    nlohmann::json templateData {};
+    auto onSuccess = [&templateData](const std::string& response)
     {
-        // Not used
+        nlohmann::json jsonResponse = nlohmann::json::parse(response);
+        templateData = jsonResponse.at("index_templates").front().at("index_template").at("template");
```

Comment:
We can avoid the use of the extra variable
```suggestion
        templateData = nlohmann::json::parse(response).at("index_templates").front().at("index_template").at("template");
```

Also, we should catch the possible exceptions here to improve the error messages down the line

```suggestion
        try
        {
            templateData = nlohmann::json::parse(response).at("index_templates").front().at("index_template").at("template");
        }
        catch (const nlohmann::json::exception& e)
        {
            logDebug1(IC_NAME, "Failed to parse index template: %s", e.what());
            throw std::runtime_error("Failed to parse index template.");
        }
```